go/datas/remote_database_handlers.go

```diff
@@ -367,46 +367,175 @@ func handleRootPost(w http.ResponseWriter, req *http.Request, ps URLParams, cs c
 		d.Panic(`Expected "last" query param value`)
 	}
 	last := hash.Parse(tokens[0])
+	// "current" should really, really be called "proposed" or something in the wire API
 	tokens = params["current"]
 	if len(tokens) != 1 {
 		d.Panic(`Expected "current" query param value`)
 	}
-	current := hash.Parse(tokens[0])
+	proposed := hash.Parse(tokens[0])
 
 	vs := types.NewValueStore(cs)
 
 	// Even though the Root is actually a Map<String, Ref<Commit>>, its Noms Type is Map<String, Ref<Value>> in order to prevent the root chunk from getting bloated with type info. That means that the Value of the proposed new Root needs to be manually type-checked. The simplest way to do that would be to iterate over the whole thing and pull the target of each Ref from |cs|. That's a lot of reads, though, and it's more efficient to just read the Value indicated by |last|, diff the proposed new root against it, and validate whatever new entries appear.
-	datasets := types.NewMap()
+	lastMap := validateLast(last, vs)
+
+	proposedMap := validateProposed(proposed, last, vs)
+	if !proposedMap.Empty() {
```

Comment:
I actually don't know, but we did it before so I just kept it. I don't know if it's a correctness thing, or if it's like..."hey, an empty map is fine and spinning up the diff machinery to figure out something we alreayd know is dumb."

I suppose I could just put this check down in the assert function, and we can take it out if we're certain it's not needed.